lib/services/user/service.go

```diff
@@ -0,0 +1,92 @@
+package user
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/jmoiron/sqlx"
+	"github.com/kennydo/artesia/lib/services"
+	"go.uber.org/zap"
+	"golang.org/x/crypto/bcrypt"
+)
+
+// DBService implements the user service, backed by the DB
+type DBService struct {
+	log *zap.SugaredLogger
+	db  *sqlx.DB
+}
+
+// GetByID gets a user by ID. Returns an error if no user is found.
+func (s *DBService) GetByID(id int) (*services.User, error) {
+	dbUser := DBUser{}
+	err := s.db.Get(&dbUser, "SELECT * FROM users WHERE id = $1", id)
+	if err != nil {
+		return nil, fmt.Errorf("No user found")
+	}
+
+	user := services.User(dbUser)
+	return &user, nil
+}
+
+// GetByEmail gets a user by their (case-insensitive) email. Returns an error if no user is found.
+func (s *DBService) GetByEmail(email string) (*services.User, error) {
+	dbUser := DBUser{}
+	err := s.db.Get(&dbUser, "SELECT * FROM users WHERE lower(email) = lower($1)", email)
+	if err != nil {
+		return nil, fmt.Errorf("No user found")
```

Comment:
Since we're using the `No user found` error more than once, let's put this value into a variable for this package:
```
var userNotFoundErr := errors.New("No user found")
```